### ISSUE TITLE
enums never have additionalProperties set

### ISSUE THREAD

**Issue by lenscas** [GREsau/schemars#46]
Seems like enums never set `additionalProperties` even if if you denied them using `#[schemars(deny_unknown_fields)]` or `#[serde(deny_unknown_fields)]`. 

My enum:
```rust
#[derive(Deserialize, Serialize, JsonSchema)]
///Every event that the game sends to the editor
#[serde(deny_unknown_fields)]
pub enum SendEvents {
    EditRectangle(AddRectangle),
    EditImage(ImageParams)
}
```
<details><summary>generated json</summary>
```json
{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "title": "SendEvents",
    "description": "Every event that the game sends to the editor",
    "anyOf": [
        {
            "type": "object",
            "required": [
                "EditRectangle"
            ],
            "properties": {
                "EditRectangle": {
                    "$ref": "#/definitions/AddRectangle"
                }
            }
        },
        {
            "type": "object",
            "required": [
                "EditImage"
            ],
            "properties": {
                "EditImage": {
                    "$ref": "#/definitions/ImageParams"
                }
            }
        }
    ],
    "definitions": {
        "AddRectangle": {
            "type": "object",
            "required": [
                "color",
                "id",
                "rectangle"
            ],
            "properties": {
                "color": {
                    "type": "string"
                },
                "id": {
                    "type": "string"
                },
                "rectangle": {
                    "$ref": "#/definitions/Rectangle"
                }
            },
            "additionalProperties": false
        },
        "ImageParams": {
            "type": "object",
            "required": [
                "id",
                "image_path",
                "location"
            ],
            "properties": {
                "id": {
                    "type": "string"
                },
                "image_path": {
                    "type": "string"
                },
                "location": {
                    "$ref": "#/definitions/Rectangle"
                }
            },
            "additionalProperties": false
        },
        "Rectangle": {
            "type": "object",
            "required": [
                "pos",
                "size"
            ],
            "properties": {
                "pos": {
                    "$ref": "#/definitions/Vector"
                },
                "size": {
                    "$ref": "#/definitions/Vector"
                }
            },
            "additionalProperties": false
        },
        "Vector": {
            "type": "object",
            "required": [
                "x",
                "y"
            ],
            "properties": {
                "x": {
                    "type": "number",
                    "format": "float"
                },
                "y": {
                    "type": "number",
                    "format": "float"
                }
            },
            "additionalProperties": false
        }
    }
}
```
</details>

the repo where this happens: https://github.com/lenscas/silver_editor/tree/feature/more_forms (the json schema's are generated by a buildscript in `/editor`, the types that are being used to generate the schema are in `/event_types/src/events.rs`

I wouldn't mind opening a pull request myself to fix this, however I haven't really made macro's before so not sure if I can fix it myself :(

Either way, thank you for making this crate. It allows me to generate typescript types with ease and simplifying a large chunk from my current project because of it!

**Comment by blitz** [comment]
I've also just run into this bug and was wondering why my schema accepts JSON that serde then does not.

**Comment by GREsau** [comment]
A fix for this has been implemented in 6a3bba1, I'll update this issue again when it's released on crates.io

**Comment by GREsau** [comment]
Fixed and published - https://crates.io/crates/schemars/0.8.1
